.github/workflows/downstream_tests.yml

```diff
@@ -50,6 +50,58 @@ jobs:
             cd altair
             mypy altair tests
 
+  marimo:
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+        os: [ubuntu-latest]
+        dependencies: ["core", "core,optional"]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+        with:
+          enable-cache: "true"
+          cache-suffix: ${{ matrix.python-version }}
+          cache-dependency-glob: "**requirements*.txt"
+      - name: clone-marimo
+        run: |
+            git clone https://github.com/marimo-team/marimo.git --depth=1
+            cd marimo
+            git log
+      - name: install-basics
+        run: uv pip install --upgrade tox virtualenv setuptools hatch --system
+      - name: install-marimo-dev
+        run: |
+            cd marimo
+            uv pip install -e ".[dev]" --system
+      - name: install-narwhals-dev
+        run: |
+            uv pip uninstall narwhals --system
+            uv pip install -e . --system
+      - name: show-deps
+        run: uv pip freeze
+      - name: Create assets directory, copy over index.html
+        run: |
+            mkdir -p marimo/marimo/_static/assets
+            cp marimo/frontend/index.html marimo/marimo/_static/index.html
+            cp marimo/frontend/public/favicon.ico marimo/marimo/_static/favicon.ico
+      - name: Run tests
+        run: |
+            cd marimo
+            hatch run test:test -v tests/ -k "not test_cli"
+            hatch run test-optional:test -v tests/ -k "not test_cli"
+        timeout-minutes: 10
+      - name: Run typechecks
+        run: |
+            cd marimo
+            hatch run typecheck:check
```

Comment:
if we ever "break" this setup due to changes in file structure or other reasons, let us know and we can create a re-usable github action that we can share to run our tests (with a `setup` script to re-install your own package)